configure.ac

```
@@ -4,7 +4,7 @@
 # - https://opensource.org/licenses/UPL
 # - <souffle root>/licenses/SOUFFLE-UPL.txt
 
-AC_INIT(souffle, m4_esyscmd([git describe --tags --always | tr -d '\n']), [souffle-talk@googlegroups.com])
+AC_INIT(souffle, m4_esyscmd([sh ./describe.sh | tr -d '\n']), [souffle-talk@googlegroups.com])
```

The describe script will fail if an out of source build is used, or the directory name changed. I think it would be better to just use 'unknown' if `git describe` is not found.
e.g.
```suggestion
AC_INIT(souffle, m4_esyscmd([(git describe --tags --always || echo unknown)| tr -d '\n']), [souffle-talk@googlegroups.com])
```

---

The proposed change will also produce "unknown" for [release sources](https://github.com/souffle-lang/souffle/releases), since they have no .git directory. Maybe this is better?

```
AC_INIT(souffle, m4_esyscmd([(git describe --tags --always || sh ./describe.sh || echo unknown)| tr -d '\n']), [souffle-talk@googlegroups.com])
```

---

The directory name is generated by github, and may easily not reflect the version in future. Without that, though, we either say the version is unknown or we start storing versions in the source code. Despite that, if the source release is being used without the git repo, then it makes sense to have some way to find a version, so we could revisit this in future. For now, though, I think we should use your suggestion of the GitHub generated name.

Given that, I would only ask for this PR that you use the configure.ac script to find the name rather than the added describe.sh script.

e.g.
```
m4_define([SOUFFLE_VERSION], m4_esyscmd([
  souffle_version=$(git describe --tags --always)
  test -z "$souffle_version" && souffle_version=$(pwd | sed '/^.*souffle-[0-9\.]*$/!d' | sed 's/^.*souffle-//')
  test -z "$souffle_version" && souffle_version="unknown"
  echo $souffle_version| tr -d '\n']))

AC_INIT(souffle, SOUFFLE_VERSION, [souffle-talk@googlegroups.com])
```
(i.e. if the dir name is not souffle-[numbers] then use "unknown")